kong/plugins/hmac-auth/migrations/postgres.lua

```diff
@@ -24,5 +24,26 @@ return {
     down = [[
       DROP TABLE hmacauth_credentials;
     ]]
+  },
+  {
+    name = "2017-06-21-132400_init_hmacauth",
+    up = function(_, _, dao)
+      local rows, err = dao.plugins:find_all { name = "hmac-auth" }
+      if err then
+        return err
+      end
+
+      for _, row in ipairs(rows) do
+        row.config.validate_request_body = false
+        row.config.enforce_headers = {}
+        row.config.algorithms = {"hmac-sha1"}
```

Comment:
This is intentional, older version only has support for `hmac-sha1`, so after migration I defaulted it to {"hmac-sha1"}. If user want to support all 4, he would need to patch the plugin.